### DISCUSSION TITLE
How to test NGRX factory selectors?

### DISCUSSION THREAD

**Discussion by Rutix** [ngrx/platform#3364] [Q&A] [answered]
Hi,

We are working on changing our selectors with props over to factory selectors as recommended here: https://github.com/ngrx/platform/issues/2980

But we are now kinda stuck how we should test our selectors now. In the past we would do something like:

`expect(foobarSelector.projector(state, { prop: '222-222-222' })).toEqual(expected);`

What we are struggling with is that in the past you could pass in a custom `state` object so you basically mock the store and only test the logic of the selector. Whats the recommended way to do this now when you have factory selectors?

Just to be clear I don't want to mock my selectors in this case because I want to test the implementation of the selector. (So I am not looking for https://github.com/ngrx/platform/discussions/3279 )

**Comment by twittwer** [comment, marked as answer]
Have you tried to invoke the factory to create the selector?
`expect(foobarSelector('222-222-222').projector(state)).toEqual(expected);`

**Reply by Rutix** [reply]
I feel stupid. I could swear I tried that before but I probably made a mistake with `)` or something. Bloody Mondays. Thanks for the fast response really appreciated!
